### PR TITLE
[TOPI] Allow conv definition to have custom kernel layout

### PR DESCRIPTION
**Pull request by vinx13**
This PR added an optional parameter `kernel_layout` to `conv`. This allows arbitrary data and kernel layout combination to be lowered from Relay, and scheduled by auto / meta schedule.

cc @junrushao1994 @masahi @tkonolige

### CODE REVIEW

**Review by tkonolige — CHANGES_REQUESTED** [review]
Looks pretty good, but I don't see any tests for different kernel_layouts. Could you add some?

**Comment by vinx13** [comment]
@tkonolige  There is no good ways to test numerical results for arbitrary combinations of layouts due to lack of schedule. Therefore, I added some tests to check `conv2d` with `data_layout == 'NCHW' and kernel_layotu == 'OIHW'` and `conv2d_nchw` have equivalent compute definition

**Comment by tkonolige** [comment]
@vinx13 I think you can use `tvm.te.create_schedule` as the schedule and the ops will run just fine. Performance might be poor but that doesn't matter for testing correctness.

File: python/tvm/topi/nn/conv2d.py [hunk 1]
```diff
@@ -57,16 +57,18 @@
 )
 
 
-def conv2d(input, filter, strides, padding, dilation, layout="NCHW", out_dtype=None):
+def conv2d(
+    input, filter, strides, padding, dilation, data_layout="NCHC", kernel_layout="", out_dtype=None
```
Comment (Ever-Kid) [inline python/tvm/topi/nn/conv2d.py:61]: maybe data_layout="NCHW"

File: python/tvm/topi/nn/conv2d.py [hunk 2]
```diff
@@ -79,17 +81,21 @@ def conv2d(input, filter, strides, padding, dilation, layout="NCHW", out_dtype=N
     dilation: int or a list/tuple of two ints
         dilation size, or [dilation_height, dilation_width]
 
-    layout : str
+    data_layout : str
         layout of data
 
+    kernel_layout : Optional[str]
+        layout of kernel. If unspecified, use default layout inferred from data_layout. "OHWI" if
+        data_layout == "NCHW", "HWIO" if data_layout == "NHWC".
```
Comment (Ever-Kid) [inline python/tvm/topi/nn/conv2d.py:89]: A little bit confused to me, shouldn’t "OIHW" & "NCHW" a default pair? I can also see this pattern at line 248.

**Pull request merged**